### DISCUSSION TITLE
Screenshots in native resolution?

### DISCUSSION THREAD

**Discussion by PrismaticRealms** [dirkwhoffmann/vAmiga#847] [Ideas]
Hello!  It's been about 4 years since I last looked into vAmiga and I am impressed with how far it's come!  I'm happy to see that it's got a beautiful, clean, intuitive user interface and seems to run software very well.  I do believe it is time for me to finally move on from FS-UAE! :)

I only have one request and it is in regards to the screenshots feature.  I would like to have the ability to take a screenshot and have the resulting PNG file be in the same resolution as the current Amiga display.  So, in other words, if I've got an NTSC game running in lores, and take a screenshot, the resulting PNG image should be 320x200 pixels in size to match the Amiga's screen resolution.

Would this be possible?

**Comment by PrismaticRealms** [comment]
Also I noticed that if the Screenshot Source setting is set to either "Visible Texture (Upscaled)" or "Entire Texture (Upscaled)", the resulting PNG file is just a black canvas.  Could be a bug?

**Comment by dirkwhoffmann** [comment]
> So, in other words, if I've got an NTSC game running in lores, and take a screenshot, the resulting PNG image should be 320x200 pixels in size to match the Amiga's screen resolution.

I think this is difficult to make, because some games switch between hires and lores modes inside a single frame and the display window's size can be rather arbitrary (both in dimension and position).

**Reply by PrismaticRealms** [reply]
I tried the latest version of vAmiga to test the screenshot functionality.  It works better!

Going back to saving screenshots in the Amiga's native resolution, if the core doesn't make it easy, then perhaps a little analysis of the texture prior to saving the image could assist here.  For example:

![Giana](https://github.com/user-attachments/assets/9f314620-fa10-4ea1-9706-5bd919f177e4)

This is a close up of one of the Giana sisters' face.  A simple visual analysis tells us that all of the pixels that make up the image are quadrupled on both axes. In other resolutions, you might find a doubling of pixels along the x-axis and quadrupled along the y.

Whatever the case might be, the screenshot feature could do some quick sampling of the texture to find out if each pixel is being repeated along the x and y axes, and then eliminate those columns/rows of pixels prior to saving the texture out to a file.  Any possibility of getting that to happen?  Would save me time in doing those steps in a paint program myself.

**Comment by dirkwhoffmann** [comment]
If we remove the border, each screenshot will have a custom size. How would this align with your use case?

**Reply by PrismaticRealms** [reply]
I would say that so long as this "Smart Trim Screenshot" (for lack of a better term) results in the consolidation of duplicated / quadruplicated pixels, meaning that every 1 pixel in the game/app translates to 1 pixel in the final screenshot, then it's better than nothing.  Even if it means the final screenshot dimensions are a bit odd.

Of course, it would be best if the screenshot process could somehow obtain the exact dimensions of the actual Amiga screen size that is being drawn to but if that information isn't available or tough to determine, then the above is a good compromise, I think.

**Comment by olafwagner** [comment]
What is the keyboard combination to take a screenshot in vAmiga? (While I can find the screenshot settings in the documentation, I can't find what the keyboard shortcuts documented. I also tried Command+3 (which is what it is on VirtualC64) but that doesn't work here).

**Comment by dirkwhoffmann** [comment]
It's Command+3 like in VirtualC64. It works here: 

<img width="912" height="810" alt="Bildschirmfoto 2026-03-27 um 19 50 30" src="https://github.com/user-attachments/assets/58a0eaaa-2367-481e-badb-f82ff5fcd885" />

Which vAmiga version did you use? 

BTW, you can also take screenshots via the Machine menu.

**Reply by olafwagner** [reply]
I am using vAmiga v4.4b8, and the screenshots option in the Machine menu is greyed out (and Cmd+3 doesn't do anything). (As an aside, I have vAmiga configured as a standard 1Mb Amiga 500 using workbench 1.3)
